sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsDeviceMemoryStore.scala

```diff
@@ -140,7 +140,12 @@ class RapidsDeviceMemoryStore(
       if (table.isDefined) {
         GpuColumnVector.from(table.get) //REFCOUNT ++ of all columns
       } else {
-        throw new UnsupportedOperationException("compressed buffer support not implemented")
+        val uncompressedBuffer = uncompressBuffer(contigBuffer, meta.bufferMeta)
```

Comment:
In practice `getColumnarBatch` needs to be called a lot less for compressed buffers. Maybe a note would be good to remind us to be careful when compression is on.

You could also change `getColumnarBatch` to be something like `getOrUncompressBatch` to make it abundantly clear.

I think the pattern will likely be: call this function and the remove the buffer, so this function should be getting called once.

Reply:
Yes, it minimally adds a reference when it's called and must be separately closed for each `getColumnarBatch` call, so I don't think we're going to see code spamming this for the same columnar batch.  I will add a node to `RapidsBuffer.getColumnarBatch` to be explicit.